service-admin/docker/app/Dockerfile

```diff
@@ -1,18 +1,23 @@
 FROM composer:2.1.11 AS composer
 
-COPY service-admin/composer.json /app/composer.json
-COPY service-admin/composer.lock /app/composer.lock
+RUN adduser -D -g '' user
+
+COPY --chown=user:user service-admin/composer.json /app/composer.json
+COPY --chown=user:user service-admin/composer.lock /app/composer.lock
 
 RUN composer install --prefer-dist --no-interaction --no-scripts --optimize-autoloader --ignore-platform-reqs
```

Comment:
Is composer still running as root here? Do we need a su command or similar to switch to the new user? I see we have `USER user` further down, but presumably this part of the docker file won't be using that? I'd also suggest using a username other than "user" as it might get confusing. Maybe "serviceuser" or "servicerunner" or "svcuser" or something?

Reply:
@townxelliot all good questions: 
- Composer is still running root here, but as this is the compile part of a multistage build this will not be a problem. This is the "baking" phase so to speak for the vendor folders.
- The `user` is used for execution time only, so we can prevent running with root privilege.
- I think we can get around the root pip install by specifying a `--user` flag - but not tested that - maybe a separate ticket to look at that (in general).
- I'll fix the name to something more meaningful - it's a first attempt to get things working.